src/plots/plots.js

```diff
@@ -1446,13 +1449,7 @@ plots.extendTrace = function(destTrace, srcTrace) {
  * The result is the original object reference with the new contents merged in.
  */
 plots.extendLayout = function(destLayout, srcLayout) {
-    return plots.extendObjectWithContainers(destLayout, srcLayout, [
-        'annotations',
-        'shapes',
-        'images',
-        'sliders',
-        'updatemenus'
-    ]);
+    return plots.extendObjectWithContainers(destLayout, srcLayout, plots.layoutArrayContainers);
```

Comment:
Yep. I thought exposing `plots.layoutArrayContainers` was a nice compromise.

Next step would be to auto-generate that list using the plot-schema. (later ...)